assets/src/application/ui/layout/entityList/filterBar/subscription/useFilterBarService.ts

```diff
@@ -18,52 +19,79 @@ const useFilterBarService: FBShook = (domain, entityListId) => {
 		service: FilterBarServiceType.FILTER,
 	});
 
-	const registerCallback = ({ callback, listId, priority, type }) => {
-		invariant(listId, 'No `listId` provided');
-		return subscribe(callback, { listId, priority, type });
-	};
-
-	const getCallbacks = ({ listId = entityListId, type }) => {
-		invariant(listId, 'No `listId` provided');
-		const allSubscriptions = getSubscriptions();
-
-		const isForList = pathEq(['options', 'listId'], listId);
-		const isOfType = pathEq(['options', 'type'], type);
-		const isOfTypeAndForList = allPass([isForList, isOfType]);
-
-		return filter(isOfTypeAndForList, allSubscriptions);
-	};
-
-	const exposeToRegistry: ExposeToRegistry = (key, value) => {
-		const existingValue = getServiceRegistryItem(key);
-		if (typeof existingValue !== 'function') {
-			addToServiceRegistry(key, value);
-		}
-	};
-
-	const getFilters = (listId = entityListId) => {
-		return getCallbacks({ listId, type: 'filter' });
-	};
-
-	const getSorters = (listId = entityListId) => {
-		return getCallbacks({ listId, type: 'sort' });
-	};
-
-	const getSearches = (listId = entityListId) => {
-		return getCallbacks({ listId, type: 'search' });
-	};
-
-	const registerFilter: FBSRegistry['registerFilter'] = (callback, priority = 10, listId = entityListId) => {
-		return registerCallback({ callback, listId, priority, type: 'filter' });
-	};
-
-	const registerSorter: FBSRegistry['registerSorter'] = (callback, priority = 10, listId = entityListId) => {
-		return registerCallback({ callback, listId, priority, type: 'sort' });
-	};
-
-	const registerSearch: FBSRegistry['registerSearch'] = (callback, priority = 10, listId = entityListId) => {
-		return registerCallback({ callback, listId, priority, type: 'search' });
-	};
+	const registerCallback = useCallback(
+		({ callback, listId, priority, type }) => {
+			invariant(listId, 'No `listId` provided');
+			return subscribe(callback, { listId, priority, type });
+		},
+		[subscribe]
+	);
+
+	const getCallbacks = useCallback(
+		({ listId = entityListId, type }) => {
+			invariant(listId, 'No `listId` provided');
+			const allSubscriptions = getSubscriptions();
+
+			const isForList = pathEq(['options', 'listId'], listId);
+			const isOfType = pathEq(['options', 'type'], type);
+			const isOfTypeAndForList = allPass([isForList, isOfType]);
```

Comment:
can't say I understand how this is working, but that's ok 🤷‍♂️

Reply:
😄 
Let me explain:
- `listId` can be `'dates-list' | 'tickets-list'`
- `type` can be `'filter' | 'sort' | 'search'`

So `isOfTypeAndForList ` checks if it's a filter for dates-list or a sorter for tickets-list and so on